pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -1147,7 +1147,7 @@ public ServiceUnitId getServiceUnitId(TopicName topicName) throws Exception {
             LOG.debug("Getting children from partitioned-topics now: {}", path);
         }
 
-        return pulsar.getLocalZkCache().getChildrenAsync(path, null).thenCompose(topics -> {
+        return pulsar.getGlobalZkCache().getChildrenAsync(path, null).thenCompose(topics -> {
```

Comment:
this change does not affect only tests and the title of the issue may suggest.
I see that in the description you cited a partial fix for a bug
can you please reflect this in the title and state it more clearly in the description ?

Reply:
of course, thanks for your suggestion.